statuspage/src/main/java/com/jagrosh/jdautilities/statuspage/Statuspage.java

```diff
@@ -0,0 +1,447 @@
+     */
+    @Nonnull
+    public AsyncFuture<ServiceStatus> getServiceStatus()
+    {
+        return get(URL_SERVICE_STATUS, this::createServiceStatus);
+    }
+
+    @Nonnull
+    protected <T> AsyncFuture<T> get(@Nonnull String url, @Nonnull Function<JSONObject, T> funtion)
+    {
+        AsyncTask<T> future = new AsyncTask<>();
+
+        // @formatter:off
+            Request request = new Request.Builder()
+                .get()
+                .url(url)
+                .header("Content-Type", "application/json")
+                .build();
+            // @formatter:on
+
+        Call call = this.client.newCall(request);
+
+        call.enqueue(new Callback()
+        {
+            @Override
+            public void onResponse(@Nonnull Call call, @Nonnull Response response)
+            {
+                ResponseBody body = response.body();
+
+                if (body == null)
+                    throw new IllegalStateException("response has no body");
+
+                Reader reader = body.charStream();
+
+                JSONObject object = new JSONObject(new JSONTokener(reader));
+
+                T t = funtion.apply(object);
+
+                future.complete(t);
+            }
+
+            @Override
+            public void onFailure(@Nonnull Call call, @Nonnull IOException e)
+            {
+                future.completeExceptionally(e);
+            }
+        });
+
+        return future;
+    }
+
+    @Nonnull
+    protected Summary createSummary(@Nonnull JSONObject object)
+    {
+        @Nonnull
+        final JSONArray componentsArray = object.getJSONArray("components");
+        @Nonnull
+        final List<Component> components = new ArrayList<>(componentsArray.length());
+        for (int i = 0; i < componentsArray.length(); i++)
+            components.add(createComponent(componentsArray.getJSONObject(i)));
+        @Nonnull
+        final JSONArray incidentsArray = object.getJSONArray("incidents");
+        @Nonnull
+        final List<Incident> incidents = new ArrayList<>(incidentsArray.length());
+        for (int i = 0; i < incidentsArray.length(); i++)
+            incidents.add(createIncident(incidentsArray.getJSONObject(i)));
+        @Nonnull
+        final JSONArray scheduledMaintenancesArray = object.getJSONArray("scheduled_maintenances");
+        @Nonnull
+        final List<ScheduledMaintenance> scheduledMaintenances = new ArrayList<>(scheduledMaintenancesArray.length());
+        for (int i = 0; i < scheduledMaintenancesArray.length(); i++)
+            scheduledMaintenances.add(createScheduledMaintenance(scheduledMaintenancesArray.getJSONObject(i)));
+        @Nonnull
+        final JSONObject pageObject = object.getJSONObject("page");
+        @Nonnull
+        final Page page = createPage(pageObject);
+        @Nonnull
+        final JSONObject statusObject = object.getJSONObject("status");
+        @Nonnull
+        final Status status = createStatus(statusObject);
```

Comment:
I don't see a reason to have `@Nonnull`, `@Nullable`, and similar kinds of annotations in method bodies.
I won't make you remove any of the method, class, and field annotations, but I would be hesitant on adding nullable annotations in the future as I have been considering adding them to the library myself, but only if and when [`feature/nullability`](https://github.com/DV8FromTheWorld/JDA/pull/587) is merged.